doc/interoperability/pandas.rst

```diff
@@ -0,0 +1,202 @@
+
+Handling Categorical data
+=========================
+
+Section to be extended.
+
+See the following references to get started:
+
+- https://scikit-learn.org/stable/glossary.html#term-categorical-feature
+- https://scikit-learn.org/stable/modules/preprocessing.html#preprocessing-categorical-features
+- https://github.com/scikit-learn-contrib/sklearn-pandas
+
+
+Dealing with heterogenous data
+==============================
+
+Many modern datasets used with Scikit-learn contain heterogenous data. For the purpose of adding bespoke preprocessing steps for separate columns, Scikit-learn provides an experimental :class:`~sklearn.compose.ColumnTransformer` API. This API (which might change in the future) allows the definition of different transformation steps to be applied to different columns in either arrays, sparse matrices or pandas DataFrames.
+
+Dealing with missing values
+===========================
+
+As per the glosary, most scikit-learn estimators do not work with missing values. If they do, NaN is the preferred representation of missing values. For more details, see https://scikit-learn.org/stable/glossary.html#term-missing-values.
+
+
+Sparse DataFrames Handling
+=============================
+
+**Issue:**
+``Sparse DataFrames`` are not automatically converted to ``scipy.sparse`` matrices.
+
+This is an issue which has vastly improved from pandas version 0.21.1 onwards. The conversation from dataframes has been largely optimized and are much faster to convert.
+
+In general, Sparse datastructures (i.e. DataFrames, Series, Arrays) are memory optimised structures of their standard counterparts. They work on the principle that they contain a lot of NaN, 0, or another repeating value (this can be specified), and as such a lot of memory can be saved, which means one can potentially work with datasets that would otherwise be too large to fit into available memory. However one has to be careful they don't get converted into the dense format by mistake.
+
+In Pandas, the sparse datastructrures are: :class:`~pandas.SparseDataFrame`, :class:`~pandas.SparseSeries` and :class:`~pandas.SparseArray`.
+The methods: :meth:`.to_sparse(fill_value=0)` and :meth:`.to_dense()` can be used to convert between normal and sparse data structures.
+The `.density` property can be called on the sparse structures to report sparseness.
+
+In scipy.sparse we have a number of various sparse matrix classes:
+
+==========  =====================================
+Class
+==========  =====================================
+bsr_matrix  Block Sparse Row matrix
+coo_matrix  Sparse matrix in COOrdinate format
+csc_matrix  Compresed Sparse Column matrix
+csr_matrix  Compresed Row matrix
+dia_matrix  Sparse matrix with diagonal storage
+dok_matrix  Dictionary of Keys based sparse matrix
+lil_matrix  Row-based linked list sparse matrix
+==========  =====================================
```

Comment:
I would remove this table, it's not really relevant. In scikit-learn we mostly use CSR / CSC providing an example of how to convert `SparseDataFrame` to CSR as done below might be enough.

Reply:
Worked on this and the 80 line length here: https://github.com/scikit-learn/scikit-learn/pull/11305/commits/37380c95de850a668ad5516a65cbc7db97c682a7